### DISCUSSION TITLE
Doubt about "How tall is this barrier?"

### DISCUSSION THREAD

**Discussion by ivanbranco** [Helium314/SCEE#631] [Q&A]
![image](https://github.com/user-attachments/assets/aa654370-dac2-4997-8e17-933668d5e252)

What would you measure for a fence with variable height? The highest past in red or the lowest part in green?
Should I use the same logic suggest for buildings (_"Use the maximum height for buildings, not the average height."_)?

**Comment by mnalis** [comment]
IMHO, you **shouldn't tag `height` there** at all, but hide that specific quest instance... 

especially if in SCEE you use AR measuring, which will add `source:height=ARcode`, which would mean it is relatively accurate, which would be even more highly misleading in cases of variable height.

But the question should much better be asked in [community tagging forum](https://community.openstreetmap.org/c/help-and-support/tagging/71), as the core of the problem is not specific to SC(EE) at all. Do link from here to there and back if you do that please, so interested parties can follow.

The core of the issue _"is there hugely predominant use case for such fence height data?"_ - for building that you mention, it is for 3D render and avoiding planes/drones/whatever hit them, so _"maximum height for buildings"_ makes sense.
But for fences and the like, even when I try hard to ignore the fact that I'm unable to find _**any realistic use**_ for it,  whichever possible use I try to invent seems to have its counterpart arguing for other point of view. e.g.:

- it might be used by urban parkour athlete, to find a route where they might jump over such barriers and avoid those where they'd be stuck in their training. Or, it might be used by police trying to corner a thief that is trying to run away in order to force him into an alley where they can't just escape by jumping over the fence.  In such cases, tagging **lowest** point would be most usable.
- but it might be used by a pre-prepared route for human drivers for model RC airplanes with high wing span urban flying competition.  Or for AI flying drones trying to stay below the radar. For that case, tagging **highest** point would be most usable, as there is no guarantee they'd be able to fit otherwise, so they'd better chose alternate route with guaranteed lower fences in advance then be stuck.

---

Anyway, here are several suggestions how it might be handled, you can use them as basis for setting up community discussion, or for making your own mind:

- just hide that quest and move on (advantage: easiest; disadvantage: other SCEE users in that area which have that quest will have to do the same, and no information is recorded). That is probably what I'd do.... 
- (only correct and precise way - **nanomapping**): split the fence in 17 (or however many) connected parts, and tag accurate height for each one (advantage: most precise and easily usable data! disadvantage: time wasted on low-value quest, database bloat)
- measure either one, and use SCEE raw tag editor to add `note=This is variable-height fence, and 2.0m is highest point; I have not measured the lowest one`. (you can also measure both and mention that in a note). Advantage: you record all data, if only some in machine-readable form. Disadvantage: more work.
- disregard the [wiki](https://wiki.openstreetmap.org/wiki/Key:height) which discourages ranges and do it anyway (that would entail manual measuring and entering it via raw tag editor e.g. `1.1 - 2.0`. I wouldn't recommend it, but there are at least several thousand of such uses (although huge majority seems to be some import). Advantage: you're not lying and are provided both pieces of information in computer-readable format. Disadvantage: even more unlikely that data consumers will use it, unless it is documented in the wiki. And more manual work.
- eyeball the height and tag the average (and deliberately coarse) `est_height=*` instead. Probably edit the quest and make it ignore elements that have `est_height` tagged too, or you'll also have to manually hide it after each such tagging.
- ask yourself _"how is this data going to be used? Is such use worthwhile investment of my time?"_, and if answer happens to be be _"no"_ for you, disable low-value quests such as that one altogether.
- invent new fancy tagging scheme for tagging specifically variable-height fences! (Like: [paving stones details](https://wiki.openstreetmap.org/wiki/Tag:surface=paving_stones); or a math function that describes the average curve encoding both maxima and minima and confidence intervals!) advantage: fun, can be arbitrarily complex and precise! disadvantage: why be so negative all the time, no need to focus on disadvantages :smile:
